### ISSUE TITLE
Push the CoreDNS 1.6.6 image in gcr.io

### ISSUE THREAD

**Issue by rajansandeep** [kubernetes/kubernetes#86219]
Need to push the latest CoreDNS image to gcr.io
The 1.6.6 image is available in docker coredns/coredns:1.6.6

 
This is the script used for pushing the image:

```
#!/bin/sh

if [ -z "$1" -o -z "$2" -o -z "$3" -o -z "$4" ]; then
    echo "usage: $0 <src-reg> <image> <tag> <dst-reg>" > /dev/stderr
    echo "example: $0 coredns coredns 1.6.6 staging-k8s.gcr.io" > /dev/stderr
    exit 1
fi

SRC_REG="$1"
IMG="$2"
TAG="$3"
DST_REG="$4"
TMP="/tmp/$RANDOM"
JSON="$TMP.json"

echo "Fetching source manifest"
docker manifest inspect ${SRC_REG}/${IMG}:${TAG} > ${JSON}

echo
echo "Pulling source images by SHA"
cat ${JSON} \
    | jq -j ".manifests[] | \"${SRC_REG}/${IMG}@\", .digest, \"\n\" " \
    | while read X; do docker pull $X; done

echo
echo "Tagging images with ARCH and OS"
cat ${JSON} \
    | jq -j ".manifests[] | \"docker tag ${SRC_REG}/${IMG}@\", .digest, \" ${DST_REG}/${IMG}:${TAG}__\", .platform.architecture, \"_\", .platform.os, \"\n\"" \
    | while read X; do $X; done

echo
echo "Pushing images"
cat ${JSON} \
    | jq -j ".manifests[] | \"gcloud docker -- push ${DST_REG}/${IMG}:${TAG}__\", .platform.architecture, \"_\", .platform.os, \"\n\"" \
    | while read X; do $X; done

echo
echo "Creating and pushing manifest"
PLATFORMS=$(cat ${JSON} | jq -j ".manifests[] | .platform.os, \"/\", .platform.architecture, \"\n\"" | paste -sd,)
echo manifest-tool push from-args \
    --platforms "${PLATFORMS}" \
    --template ${DST_REG}/${IMG}:${TAG}__ARCH_OS \
    --target ${DST_REG}/${IMG}:${TAG}
manifest-tool push from-args \
    --platforms "${PLATFORMS}" \
    --template ${DST_REG}/${IMG}:${TAG}__ARCH_OS \
    --target ${DST_REG}/${IMG}:${TAG}
```

/cc @chrisohaver 
/assign @johnbelamaric

**Comment by johnbelamaric** [comment]
Images have been pushed!

/close

**Comment by dims** [comment]
@johnbelamaric @rajansandeep next time please use the image promotion process setup by @stp-ip: 

https://github.com/kubernetes/k8s.io/pull/512/files

@stp-ip @johnbelamaric please make sure everyone who needs access to the staging / promotion process are listed in:

k8s.gcr.io/images/k8s-staging-coredns/OWNERS

**Comment by stp-ip** [comment]
The information given was that the image process is not live to push to k8s.gcr.io, which as far as I know it is still the case. Using a completely different image endpoint than before would be problematic as we will already move from k8s.gcr.io/coredns:1.6.6 to k8s.gcr.io/coredns/coredns:v1.6.6 once the final switch is done. So doing a single release change with eu.gcr.io/k8s-artifacts-prod/coredns:v1.6.6 felt like an unnecessary break.
We tested the new promotion process at the same time so we are ready once its live, but the old process was still needed so I advised creating the issue etc. My fault, if that or any of the above was incorrect.

Also in the pipeline is some automation to reduce the need for staging/promotion access as was specifically requested by Christoph Blecker.
Happy to coordinate PRs for more people having access nonetheless.
